### PR TITLE
Show file names in attachment preview

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
Show file name, file type and size in attachment preview:
*Updated test snapshots

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

### CODE REVIEW

**Comment by jwilander** [comment]
You've got a style error:
```
/home/ubuntu/jenkins/workspace/mattermost-webapp_PR-526-UGFCGOHKNYPYUM4IEZDTIJPL4AZ4X462QRY2LLHKNM2E7UKX73GQ/components/file_attachment/filename_overlay.jsx
  8:9  error  'getFileThumbnailUrl' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)
```

**Comment by dmeza** [comment]
@jwilander fixed

**Comment by lfbrock** [comment]
This looks good, I think there may be one bug where the scroll bar disappears when uploading multiple images on a small window size. 

Repro: 
1. Reduce window size 
2. Upload 5 files 

Observed: 
Can see a couple files, but can't scroll to see the others. (Tested on Chrome on Mac). 
screenshot 2018-01-02 11 15 02

Expected: 
Should be a scroll bar to scroll between the thumbnails

**Comment by lfbrock** [comment]
^ @dmeza on the above

**Comment by dmeza** [comment]
@lfbrock updated with style fixes for correct horizontal scrolling.

**Comment by hmhealey** [comment]
Code looks good, but I think we might need to do something to about how the text wraps with the close button there
image

@lfbrock Thoughts?

**Comment by lfbrock** [comment]
@hmhealey or @dmeza thoughts on how much work it would be to do something about the wrapping? 

If it's an easy fix I think that makes sense, but if it's a lot of work we have a PR to update the thumbnail design for next release anyway so we may just want to leave it for now and update it to a new design next release.

**Comment by dmeza** [comment]
@lfbrock @hmhealey fixed the file name wrapping issue. 
screen shot 2018-01-03 at 11 45 05 pm

File: components/file_attachment/filename_overlay.jsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+         */
+        handleImageClick: PropTypes.func,
+
+        /*
+         * Display in compact format
+         */
+        compactDisplay: PropTypes.bool,
+
+        /*
+         * Display in compact format
```
Comment (saturninoabril) [inline components/file_attachment/filename_overlay.jsx]: Might be a copy paste error

Comment (dmeza) [reply components/file_attachment/filename_overlay.jsx]: @saturninoabril changed.

**Comment by dmeza** [comment]
@saturninoabril updated prop description for canDownload

**Pull request merged**